### PR TITLE
Fix stuff

### PR DESCRIPTION
**Pull request by rope-hmg**
Hey, I was just passing through and I saw some opportunities for "improvement". I wasn't able to run the executables in the repo (I guess they're not Mac versions).

I don't know if you want to use any of this or not, but here it is. Use as much or as little as you like :)

### CODE REVIEW

**Comment by Mubelotix** [comment]
Wow it looks great. I will review your code tomorrow. Thank you very much for your contribution

**Comment by Mubelotix** [comment]
A bug has been introduced somewhere but I can't see where. I will see if I can fix it. Otherwise it would be easier to apply the changes step by step.

File: src/session.rs [hunk 1]
```diff
@@ -44,131 +76,95 @@ impl Session {
...
-                        result.webdriver_process = Some(p);
-                        return Ok(result);
-                    } else if let Err(e) = result {
-                        error!("Failed to create session. error : {:?}.", e);
-                        return Err(e);
-                    }
-                } else {
-                    info!{"Launching chromedriver..."}
-                    let p = Command::new("./chromedriver")
-                        .arg("--port=4444")
```
Comment (rope-hmg) [inline src/session.rs:81]: The bug might be here, it looks like I missed this argument.

**Comment by Mubelotix** [comment]
Hmm that's strange. Could you try to find and fix the bug using the tests?

**Comment by rope-hmg** [comment]
Sorry, I've been quite busy for the last week or so. Doesn't look like I will get any time for a little while yet. I will come back to this as soon as I'm free.

**Comment by Mubelotix** [comment]
Alright. I will wait for your next comments to review your PR in detail.

**Comment by Mubelotix** [comment]
I started to work on this in #4

**Pull request closed without merging**